Business Social Network/AppModules/Search/views/searchApp.js

```diff
@@ -1,8 +1,51 @@
 var search = angular.module("searchApp", []);
 
-search.controller('SearchBoxController', function($scope, $http){
+search.controller('SearchBoxController', function($scope, $location){
 
-  var userSearch;
+  //show all categories by default
+  $scope.selected = "All Categories";
+
+  //get all categories to display in dropdown menu
+  //dummy data:
+  $scope.categories = ["Clothes", "Electronics", "Books", "Bateekh"];
+  var selectedCategory = "All Categories";
+  var userSearch = ' ';
+
+  $scope.update = function(){
+
+    userSearch = $scope.search_term;
+
+    //update URL if necessary
+    if(userSearch.length > 0){
+      $location.path('/search/' + selectedCategory + '/' + userSearch);
+    }
+
+    else{
+      $location.path('/');
+    }
+
+  }
+
+  $scope.updateCategory = function(category){
+
+    $scope.selected = category;
+    selectedCategory = category;
+
+  //update URL if necessary
+    if(userSearch.length > 0){
+      $location.path('/search/' + selectedCategory + '/' + userSearch);
+    }
+
+    else{
+      $location.path('/');
+    }
+
+  }
+
+
+});
+
+search.controller('SearchResultsController', function($scope, $routeParams){
 
```

Comment:
@Swailem95 
results controller should fetch products from the database (although its not implemented yet, please handle the route on the server side to fetch all products initially then filter when user types in search results, i know you added dummy data, but you should also add the server handling for your requests in the server ProductController)